frontend/src/features/dsPreview/DatasetPreviewPage.tsx

```diff
@@ -0,0 +1,116 @@
+import BodyPreview from '../dsComponents/body/BodyPreview'
+import DatasetHeader from '../dataset/DatasetHeader'
+import NavBar from '../navbar/NavBar'
+import DatasetNavSidebar from '../dataset/DatasetNavSidebar'
+import DeployingScreen from '../deploy/DeployingScreen'
+
+import { selectSessionUserCanEditDataset } from '../dataset/state/datasetState';
+import { QriRef } from '../../qri/ref'
+
+interface DatasetPreviewPageProps {
+  qriRef: QriRef
+}
+
+const DatasetPreviewPage: React.FC<DatasetPreviewPageProps> = ({
+  qriRef
+}) => {
+  const dataset = useSelector(selectDataset)
+  const loading = useSelector(selectIsDatasetLoading)
+  const editable = useSelector(selectSessionUserCanEditDataset)
+
+  const [versionInfoContainer, { height: versionInfoContainerHeight }] = useDimensions();
+  const [expandReadme, setExpandReadme] = useState(false)
+
+  let readmeContainerHeight = versionInfoContainerHeight || 'auto'
+
+  if (expandReadme) {
+    readmeContainerHeight = 'auto'
+  }
+
+  return (
+    <div className='flex flex-col h-full w-full' style={{ backgroundColor: '#f3f4f6'}}>
+      <NavBar />
+      <div className='flex overflow-hidden w-full'>
+        <DatasetNavSidebar qriRef={qriRef} />
+        {loading
+        ? (<div className='w-full h-full p-4 flex justify-center items-center'>
+            <Spinner color='#4FC7F3' />
+          </div>)
+        : (
+          <div className='overflow-y-scroll overflow-x-hidden flex-grow relative flex'>
+            <div className='max-w-screen-lg mx-auto py-9 flex-grow'>
+              <DatasetHeader qriRef={qriRef} editable={editable} noBorder />
+              <div className='-ml-2 -mr-3 mb-5'>
+                <div className='w-7/12 px-2 inline-block align-top' style={{ height: readmeContainerHeight}}>
+                  <ContentBox className='flex flex-col h-full'>
+                    <ContentBoxTitle title='Readme'/>
+                    <div className='flex-grow overflow-hidden'>
+                      <DatasetComponent
+                        key='readme'
+                        componentName={'readme'}
+                        dataset={dataset}
+                      />
+                    </div>
+                    {!expandReadme && (<div className='text-qriblue text-sm cursor-pointer' onClick={() => { setExpandReadme(true) }}>See More</div>)}
+                  </ContentBox>
+                </div>
+                <div ref={versionInfoContainer} className='w-5/12 px-3 inline-block align-top'>
+                  <ContentBox>
+                    <div className='flex items-center border-b pb-4'>
+                      <div className='flex-grow'>
+                        <ContentBoxTitle title='Version Info' />
+                        <div className='text-qrinavy text-sm flex items-center mb-0'>
+                          <Icon icon='commit' size='sm' className='-ml-2' />
+                          <div className='font-medium'>{dataset.path.substr(dataset.path.length - 7)}</div>
```

Comment:
abstract this to a `commitishFromPath` utility function instead of hardcoding. Ideally placing it in a utility file.